core/src/main/java/hudson/model/Job.java

```diff
@@ -316,8 +318,56 @@ public boolean isBuilding() {
      */
     public boolean isLogUpdated() {
         RunT b = getLastBuild();
-        return b!=null && b.isLogUpdated();
-    }    
+        return b != null && b.isLogUpdated() || isCurrentWorkUnit();
+    }
+
+    /**
+     * Returns {@code true} if the job is a current work unit on any node.
+     *
+     * @return {@code true} if the job is a current work unit on any node.
+     * @since FIXME
+     */
+    public boolean isCurrentWorkUnit() {
+        if (this instanceof SubTask) {
+            // if you are not a SubTask then you cannot be a work unit on any node.
+            final Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins != null) {
+                if (isCurrentWorkUnit(jenkins)) {
+                    return true;
+                }
+                for (Node n : jenkins.getNodes()) {
+                    if (isCurrentWorkUnit(n)) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns {@code true} if the job is a current work unit on the specified node.
+     *
+     * @param node the specified node
+     * @return {@code true} if the job is a current work unit on the specified node.
+     */
+    private boolean isCurrentWorkUnit(Node node) {
+        final Computer c = node.toComputer();
+        if (c == null) return false;
```

Comment:
There is a risk of false-negative result if nodes are being reconfigured in parallel with checks.